providers/gce/gce_instances.go

```diff
@@ -551,19 +551,24 @@ func (g *Cloud) AliasRangesByProviderID(providerID string) (cidrs []string, err
 		for _, r := range networkInterface.AliasIpRanges {
 			cidrs = append(cidrs, r.IpCidrRange)
 		}
-		ipv6Addr := getIPV6AddressFromInterface(networkInterface)
-		if ipv6Addr != "" {
-			// The podCIDR range is the first /112 subrange from the /96 assigned to
-			// the node
-			ipv6PodCIDR := fmt.Sprintf("%s/112", ipv6Addr)
-			cidrs = append(cidrs, ipv6PodCIDR)
-		} else {
-			klog.Infof("No IPv6 addresses found for %s", providerID)
-		}
+		cidrs = g.AccommodateIPV6Addresses(cidrs, networkInterface, providerID)
 	}
 	return
 }
 
+func (g *Cloud) AccommodateIPV6Addresses(cidrs []string, networkInterface *compute.NetworkInterface, providerID string) []string {
+	ipv6Addr := getIPV6AddressFromInterface(networkInterface)
+	if ipv6Addr != "" {
+		// The podCIDR range is the first /112 subrange from the /96 assigned to
+		// the node
+		ipv6PodCIDR := fmt.Sprintf("%s/112", ipv6Addr)
```

Comment:
if this return an IP address you may want to canonalize it,  example with IPv4 addresses for simplicity, we don't want to return "192.168.0.23/24", do we? we should return 192.168.0.0/24 no?

if you convert to net.IP you can mask the ip and obtain the subnet in canonical format